src/client/mesh_generator_thread.cpp

```diff
@@ -97,6 +85,7 @@ bool MeshUpdateQueue::addBlock(Map *map, v3s16 p, bool ack_block_to_server, bool
 					if (block) {
 						block->refGrab();
 						q->map_blocks[i] = block;
+						q->data->fillBlockData(pos, block->getData());
```

Comment:
If `if (!q->map_blocks[i]) {` is not true, that is a block did exist already, the data needs to be updated.

Reply:
That should not happen, the initialization logic below grabs a reference, so the map is not allowed to unload the block.

Reply:
Why unload?
The nodes can have changed, so it's resubmitted.